packages/ember-data/lib/system/relationships/state/has-many.js

```diff
@@ -84,12 +85,12 @@ ManyRelationship.prototype.removeRecordFromOwn = function(record, idx) {
 };
 
 ManyRelationship.prototype.notifyRecordRelationshipAdded = function(record, idx) {
-  var type = this.relationshipMeta.type;
-  Ember.assert("You cannot add '" + record.constructor.typeKey + "' records to the " + this.record.constructor.typeKey + "." + this.key + " relationship (only '" + this.belongsToType.typeKey + "' allowed)", (function () {
+  var type = this.store.modelFor(this.relationshipMeta.type);
```

Comment:
Should this also use typeForRelationshipMeta instead of trying to directly access the type?